internal/config/endpoints.go

```diff
@@ -14,17 +14,54 @@
 
 package config
 
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
 // Endpoint represents a Cloud Run service deploy in a particular region.
 type Endpoint struct {
 	// URL is the HTTPS URL of the service
 	URL string
 	// Region is the programmatic name of the region where the endpoint is
-	// deloyed, e.g., us-central1.
+	// deployed, e.g., us-central1.
 	Region string
 	// RegionName is the geographic name of the region, e.g., Iowa.
 	RegionName string
 }
 
+// GetEndpointsFromServer is used by the cli to generate an Endpoint map
+// using json served by the gcping endpoints.
+func GetEndpointsFromServer(ctx context.Context, endpointsURL string) (map[string]Endpoint, error) {
```

Comment:
Getters are usually written as just the thing they're getting, e.g., EndpointsFromServer.